backend/src/routes/getOtp.ts

```diff
@@ -0,0 +1,97 @@
+import { otp, registration } from '@/db/schema';
+import { sendMessage } from '@/utils/sms';
+import { desc, eq } from 'drizzle-orm';
+import type { FastifyInstance } from 'fastify';
+import otpGenerator from 'otp-generator';
+import * as schema from '../db/schema';
+
+interface GetOtpRequestBody {
+  phone_number: string;
+}
+
+// OTP validity duration : 15 mins
+const OTP_VALIDITY_TIME = 900;
+
+// as + symbol is not taken into query params
+// - + will be added after the number is processed
+export function getOtp(fastify: FastifyInstance) {
+  fastify.post<{ Body: GetOtpRequestBody }>(
+    '/get_otp',
+    {
+      schema: {
+        body: {
+          type: 'object',
+          required: ['phone_number'],
+          properties: {
+            phone_number: { type: 'string', pattern: '^\\+[1-9]\\d{1,14}$' },
+          },
+        },
+      },
+    },
+    async (request, reply) => {
+      try {
+        const { phone_number } = request.body as { phone_number: string };
+
+        // validating if phone number exists in db
+        const record_phone_number = await fastify.db.query.registration
+          .findFirst({
+            where: eq(registration.phone_number, phone_number),
+          })
+          .execute();
+
+        if (!record_phone_number) {
+          return reply.code(500).send({ message: 'No record exists with current phone number' });
```

Comment:
Error could should probably be `40x` (400) since it's a request error, not a server error.